src/xwayland/XWM.hpp

```diff
@@ -58,6 +57,52 @@ struct SXSelection {
     std::unique_ptr<SXTransfer> transfer;
 };
 
+class XCBConnection {
+  public:
+    XCBConnection(int fd) {
+        connection = xcb_connect_to_fd(fd, nullptr);
+    }
+
+    ~XCBConnection() {
+        if (connection) {
+            xcb_disconnect(connection);
+        }
+    }
+
+    bool hasError() const {
+        return xcb_connection_has_error(connection);
+    }
+
+    xcb_connection_t* get() const {
```

Comment:
instead of all these gets why not just add an implicit cast to `xcb_connection_t*`?

Reply:
Is just good practices to make explicit to we are retrieving the underlying raw pointer  without transferring ownership. I suppose since in this case we are just managing the connection and not doing anything crazy in the class that could result in unintended copies using the implicit cast is a good solution too.

Reply:
fair enough